### PR TITLE
pybind/mgr: Add packaging module

### PR DESCRIPTION
**Pull request by Matan-B**
Introduced: 9f0ee9798563efe4ce9759f5d8c0ca6aa47a6838

`ModuleNotFoundError: No module named 'packaging'`  was spotted in few teuthology log runs.
(https://pulpito.ceph.com/yuvalif-2022-12-23_07:38:27-rgw-wip-yuval-test-main-distro-default-smithi/7126208/)

Fixes: https://tracker.ceph.com/issues/58385

Signed-off-by: Matan Breizman <mbreizma@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [x] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/pybind/mgr/requirements-required.txt [hunk 1]
```diff
@@ -14,3 +14,4 @@ requests-mock
 scipy
 werkzeug
 natsort
+packaging
```
Comment (tchaikov) [inline src/pybind/mgr/requirements-required.txt:17]: please note the rationales in the commit message, and would be better to note the offending commit in the commit message also.

**Review by tchaikov — APPROVED** [review]

**Comment by tchaikov** [comment]
@Matan-B sorry for introducing the failure. could you help check if the dependencies of`python3-packaging` on el8 and `python3-packaging` on ubuntu are added accordingly?

**Comment by tchaikov** [comment]
the former should be listed as a `Requires` of `mgr-modules-core`, as `ceph/mgr/prometheus` is packaged as part of `mgr-modules-core` . while the latter should be listed as a dependency of `ceph-mgr-modules-core` on debian/ubuntu.

**Comment by Matan-B** [comment]
> the former should be listed as a `Requires` of `mgr-modules-core`, as `ceph/mgr/prometheus` is packaged as part of `mgr-modules-core` . while the latter should be listed as a dependency of `ceph-mgr-modules-core` on debian/ubuntu.

@tchaikov Thank you for the clarification.

File: debian/control [hunk 2]
```diff
@@ -107,6 +107,7 @@ Build-Depends: automake,
                python3-venv,
                python3-werkzeug <pkg.ceph.check>,
                python3-yaml,
+               python3-packaging,
```
Comment (tchaikov) [inline debian/control:110]: python3-packaing is not a build dependency. it is a runtime dependency.

**Review by tchaikov — CHANGES_REQUESTED** [review]

File: debian/control (hunk 2 above)
Comment (tchaikov) [reply debian/control:110]: and the commit message does not match with the change.

Comment (Matan-B) [reply debian/control]: @tchaikov Fixed.

**Review by tchaikov — APPROVED** [review]

**Comment by Matan-B** [comment]
jenkins test make check

**Comment by Matan-B** [comment]
Closing in favor of #49712 49712

**Comment by Matan-B** [comment]
Closing in favor of #49712

**Pull request closed without merging**